### PR TITLE
Disable compression to be a configurable feature

### PR DESCRIPTION
**Pull request by vibhansa-msft**
Fix for #1100

### CODE REVIEW

File: component/azstorage/config.go [hunk 1]
```diff
@@ -464,9 +465,9 @@ func ParseAndValidateConfig(az *AzStorage, opt AzStorageOptions) error {
 		log.Warn("unsupported v1 CLI parameter: debug-libcurl is not applicable in blobfuse2.")
 	}
 
-	log.Info("ParseAndValidateConfig : Account: %s, Container: %s, AccountType: %s, Auth: %s, Prefix: %s, Endpoint: %s, ListBlock: %d, MD5 : %v %v, Virtual Directory: %v",
+	log.Info("ParseAndValidateConfig : Account: %s, Container: %s, AccountType: %s, Auth: %s, Prefix: %s, Endpoint: %s, ListBlock: %d, MD5 : %v %v, Virtual Directory: %v, Compression: %v",
```
Comment (souravgupta-msft) [inline component/azstorage/config.go:468]:
```suggestion
	log.Info("ParseAndValidateConfig : Account: %s, Container: %s, AccountType: %s, Auth: %s, Prefix: %s, Endpoint: %s, ListBlock: %d, MD5 : %v %v, Virtual Directory: %v, Disable Compression: %v",
```

**Review by souravgupta-msft — APPROVED** [review]

File: setup/baseConfig.yaml [hunk 2]
```diff
@@ -144,7 +144,7 @@ azstorage:
   update-md5: true|false <set md5 sum on upload. Impacts performance. works only when file-cache component is part of the pipeline>
   validate-md5: true|false <validate md5 on download. Impacts performance. works only when file-cache component is part of the pipeline>
   virtual-directory: true|false <support virtual directories without existence of a special marker blob>
-
+  disable-compression: true|false <disable transport layer content encoding like gzip, set this flag to true if blobs have content-encoding set in container>
```
Comment (gapra-msft) [inline setup/baseConfig.yaml:147]: Should we add a test for this feature?

Comment (vibhansa-msft) [reply setup/baseConfig.yaml:147]: we can just add a cli validation kind of a test. Data validation kind of thing can not be done as this goes to http client options and we do not have any direct control on testing it.

Comment (vibhansa-msft) [reply setup/baseConfig.yaml:147]: Added config test and moved block blob based proxy test in code coverage to use disable-compression

**Pull request merged**